test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -1189,42 +1244,12 @@ class test_ucp_sockaddr_wireup_fail : public test_ucp_sockaddr_wireup {
         EXPECT_TRUE(test_all_ep_flags(e, wait_ep_flags) ||
                     (sender().get_err_num() > 0));
 
-        if (fail_wireup_type == FAIL_WIREUP_MSG_ADDR_PACK) {
-            check_p2p_lanes(e, wait_ep_flags);
-
-            /* Emulate failure of preparation of WIREUP MSG sending by setting
-             * the device address getter to the function that always returns
-             * error */
-            UCS_ASYNC_BLOCK(&worker->async);
-            for (ucp_rsc_index_t iface_id = 0; iface_id < worker->num_ifaces;
-                 ++iface_id) {
-                ucp_worker_iface_t *wiface = worker->ifaces[iface_id];
-
-                wiface->iface->ops.iface_get_device_address =
-                        reinterpret_cast<uct_iface_get_device_address_func_t>(
-                                ucs_empty_function_return_ep_timeout);
+        if (fail_wireup_type != FAIL_WIREUP_MSG_SEND) {
+            emulate_failure(e, fail_wireup_type);
+            if ((fail_wireup_type == FAIL_WIREUP_CONNECT_TO_EP) ||
```

Comment:
maybe call wait_ep_err_or_wireup_msg_done() in line 1255, instead of wait_for_flag?
then we don't need this if here

Reply:
done